fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -2643,7 +2653,13 @@ public void createDb(CreateDbStmt stmt) throws DdlException {
                 }
             } else {
                 id = getNextId();
-                Database db = new Database(id, fullDbName);
+                if (engineName.equals("builtin")) {
+                    db = new Database(id, fullDbName);
+                } else if (engineName.equals("iceberg")) {
+                    db = createIcebergDb(stmt, id, icebergTables);
```

Comment:
`createIcebergDb` may be a time cost methods? but it is executed with lock holding.